yb-voyager/src/query/sqltransformer/sqlfiletransformer.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sqltransformer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	pg_query "github.com/pganalyze/pg_query_go/v6"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/constants"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/query/queryparser"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils/sqlname"
+)
+
+type IndexFileTransformer struct {
+	skipPerformanceOptimizations bool
+	sourceDBType                 string
+	redundantIndexesToRemove     map[string]string
```

Comment:
this is actually a map of redundant->existing, right? 
can we be explicit about that in the variable name, i got a little confused. 
`redundantIndexToExistingIndexMap`?